### ISSUE TITLE
Define function type for root resolvers of queries, mutations, and subscriptions.

### ISSUE THREAD

**Issue by zephraph** [avantcredit/gql2ts#59]
Let's say I have a schema something like this...

```graphql
type Query {
  user(userId: Int!): User! 
}

type User { ... }

schema {
  query: Query
}
```

I would like to generate the type definition of the actual resolver functions themselves for queries, mutations, and subscriptions. 

```typescript
interface IQuery {
    user(parent: any, args: { userId: number }, context: any, info: any): User;
}
```

This would allow simple typechecking of resolvers (or other things)...

```typescript
const query: IQuery = {
  user(_, { userId }): User { /* ... */ } 
}
```

This way if your schema file changes but someone forgets to update the resolvers, mutations, etc then you're covered. 

Thoughts?

**Comment by brettjurgens** [comment]
can totally see the utility in this.

will probably be able to utilize the new custom language/formatter features from the 1.0.0 release

i'm gonna sleep on this and come back to it tomorrow morning

**Comment by zephraph** [comment]
Any thoughts? I'd be willing to help out, put in a PR. Could just a nudge in the right direction though.

**Comment by brettjurgens** [comment]
yep, I'm not sure it can be done as-is with just a format package.

but I think it should be semi-easy, given the `from-schema` code. 

unfortunately, I'm moving this week and have limited time until around wednesday, when I'll have time to take another look at it.

definitely feel free to get started w/ a PR if you'd like, if not, I'll for sure get around to it on wednesday

**Comment by brettjurgens** [comment]
@zephraph haven't forgotten about this, will start on a PR today 😄

**Comment by brettjurgens** [comment]
this is closed in #148
